crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -108,6 +128,24 @@ pub struct Polyline2d<const N: usize> {
 }
 impl<const N: usize> Primitive2d for Polyline2d<N> {}
 
+impl<const N: usize> FromIterator<Vec2> for Polyline2d<N> {
+    fn from_iter<I: IntoIterator<Item = Vec2>>(iter: I) -> Self {
+        let mut vertices: [Vec2; N] = [Vec2::ZERO; N];
+
+        for (index, i) in iter.into_iter().enumerate() {
```

Comment:
We want to be sure that we only fetch the first `N` elements of the iterator. This will panic with an out-of-bounds error on longer iterators currently.

Reply:
I made sure that this won't panic anymore by checking if the current index is greater than or equal to `N`. Is there a more efficient/cleaner way to do this or is it fine as is?

Reply:
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take is nicer :)

Reply:
Ok I implemented `iter.take()`, is there anything else?